app/move/controllers/view.js

```diff
@@ -16,22 +19,24 @@ module.exports = function view(req, res) {
       ? cancellationComments
       : req.t(`fields::cancellation_reason.items.${cancellationReason}.label`)
   const bannerStatuses = ['cancelled']
+
+  const updateUrls = getUpdateUrls(updateSteps, move.id, res.locals.canAccess)
+  const updateLinks = getUpdateLinks(updateSteps, req.t, updateUrls)
+
   const urls = {
-    update: {},
+    update: updateUrls,
   }
-  updateSteps.forEach(step => {
-    const url = Object.keys(step)[0].replace(/^\//, '')
-    const key = url.replace(/-/g, '_')
-    urls.update[key] = `/move/${move.id}/edit/${url}`
-    // some assessment categories aren't suffixed with _information
-    if (key.endsWith('_information')) {
-      const keyWithoutInformation = key.replace(/_information$/, '')
-      urls.update[keyWithoutInformation] = urls.update[key]
+
+  const metaListActions = {}
+  Object.keys(updateLinks).forEach(key => {
+    metaListActions[key] = {
+      classes: 'app-meta-list__action--sidebar',
```

Comment:
It feels like all of this should live in the presenter. It's purely related to how we display the actions when they're in that component so feels better to contain them there rather than in this controller.

This controller is starting to contain a lot of logic and we've previously tried (where possible) to keep thin controllers and extract any complicated logic to smaller modules.

Reply:
Done